guake/guake_app.py

```diff
@@ -1004,6 +1004,17 @@ def is_using_unity(self):
                 return True
         return False
 
+    def is_using_gnome(self):
+        linux_distrib = platform.linux_distribution()
+        if float(linux_distrib[1]) - 0.01 < 11.10:
+            if os.environ.get('XDG_CURRENT_DESKTOP', '').lower() == "gnome".lower():
+                return True
+        else:
+            if os.environ.get('XDG_CURRENT_DESKTOP', '').lower() == "ubuntu:gnome".lower():
```

Comment:
`"ubuntu:gnome".lower()` is the same as just `"ubuntu:gnome"`; likewise for `"gnome"`

Reply:
I understand that you copied the version comparison logic from Gaetan's Unity-related code. Nevertheless, **casting version strings to float** isn't going to fly very far; that's simply incorrect. With versions, it isn't true that `11.2 > 11.10`. Either add a big fat `FIXME` right here (and there with Unity), or take e.g. [semver lib](https://pypi.org/project/semver/) and do it right.

Reply:
Next piece of copy-paste. Could you make a variable here?
```python
def is_using_gnome(self):
    linux_distrib = platform.linux_distribution() [1]
    xdg_desktop = os.environ.get('XDG_CURRENT_DESKTOP', '').lower()
    return (distro_before_11_10 and xdg_desktop == "gnome"
          or distro_after_11_10 and xdg_desktop == "ubuntu:gnome")
```

Reply:
As for version comparison, casting to float is not a problem for Ubuntu as its version number always have a trailing zero in case of one-digit number (so it would be 11.02, not 11.2.)

Anyway, semver has problems with trailing zeroes (it raises a ValueError exception).
I found this [packaging.version](https://packaging.pypa.io/en/latest/version/#packaging.version.parse) module that works just fine (11.04 < 11.10 yields True, 11.2 > 11.10 yields False)

I assume `packaging` should be added to the requirements, but I'm not sure how this is handled.
Should I add this line:  
```packaging = "*"```
to `Pipfile` ?

Reply:
As for the variable, I think I made a mistake in using `XDG_CURRENT_DESKTOP` in both cases.
It should be `DESKTOP_SESSION` if version is lower than 11.10.

Also, I think that `linux_distrib[1]) - 0.01` was actually meant to actually include 11.10, so I'm changing `<` for `<=`.

Reply:
I pushed changes to `is_using_gnome`. I'll wait for your feedback to resolve the conversation